src/allmydata/storage/server.py

```diff
@@ -431,9 +413,14 @@ def bucket_writer_closed(self, bw, consumed_size):
         if self.stats_provider:
             self.stats_provider.count('storage_server.bytes_added', consumed_size)
         del self._bucket_writers[bw.incominghome]
-        if bw in self._bucket_writer_disconnect_markers:
-            canary, disconnect_marker = self._bucket_writer_disconnect_markers.pop(bw)
-            canary.dontNotifyOnDisconnect(disconnect_marker)
+        for handler in self._call_on_bucket_writer_close:
```

Comment:
curious why these are "bare" callbacks (instead of e.g. Deferred()s since we're using twisted anyway).